build/cmake/DefineOptions.cmake

```diff
@@ -164,31 +164,36 @@ message(STATUS "Thrift package version:                       ${PACKAGE_VERSION}
 message(STATUS "Build configuration Summary")
 message(STATUS "  Build Thrift compiler:                      ${BUILD_COMPILER}")
 message(STATUS "  Build compiler plugin support:              ${WITH_PLUGIN}")
+if (WITH_PLUGIN_COMPILER_TOO_OLD)
+  MESSAGE_DEP(WITH_PLUGIN "Disabled due to older compiler")
+else()
+  MESSAGE_DEP(WITH_PLUGIN "Disabled by WITH_PLUGIN=OFF")
```

Comment:
I found that in the CMake environment, WITH_PLUGIN is honored properly.  It's in the autoconf environment where it seems to be messed up.